python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+            uniq[covered] = 4*(4**uniq_order) + pix_shift[covered]
+
+        # Remove duplicate pixels
+        uniq = np.unique(uniq)
+
+        # Output to fits
+        tbl = np.zeros(uniq.size, dtype=[('UNIQ', 'i8')])
+        tbl['UNIQ'] = uniq
+
+        order = np.log2(tbl['UNIQ']//4).astype(np.int32)//2
+        moc_order = np.max(order)
+
+        hdu = fits.BinTableHDU(tbl)
+        hdu.header['PIXTYPE'] = 'HEALPIX'
+        hdu.header['ORDERING'] = 'NUNIQ'
+        hdu.header['COORDSYS'] = 'C'
+        hdu.header['MOCORDER'] = moc_order
+        hdu.header['MOCTOOL'] = 'lsst.pipe.tasks.hips.GenerateHipsTask'
+
+        uri = hips_base_path.join("Moc.fits")
+
+        with ResourcePath.temporary_uri(suffix=uri.getExtension()) as temporary_uri:
+            hdu.writeto(temporary_uri.ospath)
+
+            uri.transfer_from(temporary_uri, transfer="copy", overwrite=True)
+
+
+class GenerateColorHipsConnections(pipeBase.PipelineTaskConnections,
+                                   dimensions=("instrument", ),
+                                   defaultTemplates={"coaddName": "deep"}):
+    hips_exposure_handles = pipeBase.connectionTypes.Input(
+        doc="HiPS-compatible HPX images.",
+        name="{coaddName}Coadd_hpx",
+        storageClass="ExposureF",
+        dimensions=("healpix11", "band"),
+        multiple=True,
+        deferLoad=True,
+    )
+
+
+class GenerateColorHipsConfig(GenerateHipsConfig,
+                              pipelineConnections=GenerateColorHipsConnections):
+    """Configuration parameters for GenerateColorHipsTask."""
+    blue_channel_band = pexConfig.Field(
+        doc="Band to use for blue channel of color pngs.",
+        dtype=str,
+        default='g',
+    )
+    green_channel_band = pexConfig.Field(
+        doc="Band to use for green channel of color pngs.",
+        dtype=str,
+        default='r',
+    )
+    red_channel_band = pexConfig.Field(
+        doc="Band to use for red channel of color pngs.",
+        dtype=str,
+        default='i',
+    )
+    png_color_asinh_minimum = pexConfig.Field(
+        doc="AsinhMapping intensity to be mapped to black for color png scaling.",
+        dtype=float,
+        default=0.0,
+    )
+    png_color_asinh_stretch = pexConfig.Field(
+        doc="AsinhMapping linear stretch for color png scaling.",
+        dtype=float,
+        default=5.0,
+    )
+    png_color_asinh_softening = pexConfig.Field(
+        doc="AsinhMapping softening parameter (Q) for color png scaling.",
+        dtype=float,
+        default=8.0,
+    )
+
+
+class GenerateColorHipsTask(GenerateHipsTask):
+    """Task for making a HiPS tree with color pngs."""
+    ConfigClass = GenerateColorHipsConfig
+    _DefaultName = "generateColorHips"
+
+    @timeMethod
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        dims = inputRefs.hips_exposure_handles[0].dataId.names
```

Comment:
I'd rather this code duplication was removed. It seems to be the same as the parent class runQuantum until the data_bands turns up and even that code could be shared with the parent class if the parent class runQuantum then checked that there was only one band in data_bands and it matched the quantum dataId.